tracker/event_tracker.go

```diff
@@ -99,10 +114,23 @@ func (e *EventTracker) Start(ctx context.Context) error {
 	}()
 
 	go func() {
+		defer wg.Done()
+
 		if err := tt.Sync(ctx); err != nil {
-			e.logger.Error("failed to sync", "error", err)
+			errCh <- fmt.Errorf("failed to sync: %w", err)
+
+			return
 		}
 	}()
 
+	wg.Wait()
```

Comment:
I think there is a downstream dependancy which this wait precludes

Reply:
Can we use `errgroup` instead of `waitgroup` + `channel` https://pkg.go.dev/golang.org/x/sync/errgroup ?

Reply:
Also, I am not sure... will this block server if there is no error at all?